compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPU2DScalableTo1DScalable.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TileSizeSelection.h"
+#include "iree/compiler/Codegen/LLVMCPU/PassDetail.h"
+#include "iree/compiler/Codegen/LLVMCPU/Passes.h"
+#include "iree/compiler/Codegen/LLVMCPU/Utils.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "mlir/Dialect/SCF/Utils/Utils.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir::iree_compiler {
+
+namespace {
+
+/// Currently, IREE requires `lowering_config`s to be propagated to all compute
+/// ops within a dispatch region. This can be problematic for SME, as it does
+/// not make sense to use 2D-scalability for all ops within a dispatch region.
+/// The Arm SME target only supports 2D scalable outer products, so if it's not
+/// an outer product, we can only scalably vectorize in one dimension.
```

Comment:
[nit] I would avoid terms like "make sense" in docs - they tend to be quite subjective 😅 Instead, lets keep this short and focus on the ISA aspects that drive the need for this.

```suggestion
/// ops within a dispatch region. This can be problematic for SME which only supports
// 2D scalable outer product operations - if an operation cannot be lowered to an
/// outer product, we can only scalably vectorize in one dimension.
```

Also, is it really "outer-product" that's key here or is it the fact that we need something that works with `ZA`? It just happens that today we only have `linalg.matmul` + `linalg.fill` that we lower to outer-products. But this list is likely to expand.

Reply:
I think outer product is key. There's other operations that can use ZA, but AFAIK only the outer product instructions are scalable in two dimensions. Other ZA instructions deal with vector groups, which would only be scalable in one dimension.